UserTools/VtxSeedFineGrid/VtxSeedFineGrid.cpp

```diff
@@ -0,0 +1,470 @@
+		myFoMCalculator.TimePropertiesLnL(meantime, timefom);
+		myFoMCalculator.ConePropertiesFoM(coneAngle, conefom);
+		fom = timefom * 0.5 + conefom * 0.5;
+		std::cout << "seed (" << seedX<<","<<seedY<<","<<seedZ<<") fom= " << fom << " best= " << bestFOM[0] << endl;
+
+		if (fom > bestFOM[0]) {
+			if (multiGrid) {
+				bestFOM[2] = bestFOM[1];
+				bestFOM[1] = bestFOM[0];
+				centerIndex[2] = centerIndex[1];
+				centerIndex[1] = centerIndex[0];
+			}
+			bestFOM[0] = fom;
+			peakX = seedX;
+			peakY = seedY;
+			peakZ = seedZ;
+			SeedDir = iSeed.GetDirection();
+			thisCenterSeed = iSeed;
+			centerIndex[0] = m;
+		}
+		if (multiGrid) {
+			if (fom < bestFOM[0] && fom > bestFOM[1]) {
+				bestFOM[2] = bestFOM[1];
+				centerIndex[2] = centerIndex[1];
+				bestFOM[1] = fom;
+				peakX = seedX;
+				peakY = seedY;
+				peakZ = seedZ;
+				SeedDir = iSeed.GetDirection();
+				thisCenterSeed = iSeed;
+				centerIndex[1] = m;
+			}
+			if (fom < bestFOM[0] && fom<bestFOM[1] && fom>bestFOM[2]) {
+				bestFOM[2] = fom;
+				peakX = seedX;
+				peakY = seedY;
+				peakZ = seedZ;
+				SeedDir = iSeed.GetDirection();
+				thisCenterSeed = iSeed;
+				centerIndex[2] = m;
+			}
+		}
+	}
+	//	return thisCenterSeed.GetPosition();
+
+	}
+	return thisCenterSeed.GetPosition();
+}
+
+void VtxSeedFineGrid::GenerateFineGrid() {
+	Position Seed;
+	RecoVertex thisFineSeed;
+	vSeedVtxList->clear();
+	double medianTime;
+	//double length = NSeeds something.  TODO for now setting to standard size 25x25x25, with seeds 5cm apart.
+	if (multiGrid) {
+		for (int l = 0; l < 3; l++) {
+			for (int i = 0; i < 5; i++) {
+				for (int j = 0; j < 5; j++) {
+					for (int k = 0; k < 5; k++) {
+						Seed.SetZ(Center.at(l).Z() - 10 + 5 * i);
+						Seed.SetX(Center.at(l).X() - 10 + 5 * j);
+						Seed.SetY(Center.at(l).Y() - 10 + 5 * k);
+
+
+						//medianTime = this->GetMedianSeedTime(Seed);
+						thisFineSeed.SetVertex(Seed, medianTime);
+						thisFineSeed.SetDirection(SeedDir);
+						vSeedVtxList->push_back(thisFineSeed);
+
+					}
+				}
+			}
+		}
+	}
+	else {
+		int l = 0;
+		for (int i = 0; i < 5; i++) {
+			for (int j = 0; j < 5; j++) {
+				for (int k = 0; k < 5; k++) {
+					Seed.SetZ(Center.at(l).Z() - 10 + 5 * i);
+					Seed.SetX(Center.at(l).X() - 10 + 5 * j);
+					Seed.SetY(Center.at(l).Y() - 10 + 5 * k);
+
+
+					//medianTime = this->GetMedianSeedTime(Seed);
+					thisFineSeed.SetVertex(Seed, medianTime);
+					thisFineSeed.SetDirection(SeedDir);
+					vSeedVtxList->push_back(thisFineSeed);
+
+				}
+			}
+		}
+	}
+}
+
+Direction VtxSeedFineGrid::findDirectionMRD() {
+	std::vector<BoostStore>* Tracks;
+	m_data->Stores["MRDTracks"]->Get("MRDTracks", Tracks);
+	m_data->Stores["MRDTracks"]->Get("NumMrdTracks", numtracksinev);
+
+	if (numtracksinev > 1) Log("Multiple tracks need work; just using first for now", v_debug, verbosity);
+	double gradx, grady, theta, phi;
+	Direction startVertex, endVertex, result;
+	BoostStore* thisTrack = &(Tracks->at(0));

```

Comment:
as before this will throw an exception if there are no MRDTracks

Reply:
Which demonstrates the point that when code is duplicated, bugs in that code will need to be fixed in multiple locations. Please consolidate this function with the other instance(s).